jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2522,6 +2522,12 @@ public boolean startsWith(String prefix) {
 	 *				when prefix is null
 	 */
 	public boolean startsWith(String prefix, int start) {
+		if (prefix.length()==1) {
```

Comment:
Please use spaces around binary operators (e.g. `==`) in both versions of this method.